src/box/txn_limbo.c

```diff
@@ -451,14 +451,16 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
 	struct txn_limbo_entry *e, *tmp;
 	rlist_foreach_entry_safe_reverse(e, &limbo->queue,
 					 in_queue, tmp) {
+		bool needs_wakeup = e->txn->fiber != NULL;
 		e->txn->signature = TXN_SIGNATURE_QUORUM_TIMEOUT;
 		e->txn->limbo_entry = NULL;
 		txn_limbo_abort(limbo, e);
 		txn_clear_flags(e->txn, TXN_WAIT_SYNC | TXN_WAIT_ACK);
 		txn_limbo_complete(e->txn, false);
 		if (e == entry)
 			break;
-		fiber_wakeup(e->txn->fiber);
+		if (needs_wakeup)
+			fiber_wakeup(e->txn->fiber);
```

Comment:
Do we really need `fiber_wakeup` here? `txn_complete_fail` will call `txn_free_or_wakeup`, which will wake up the fiber, if it's not `NULL`. All tests seem to pass for me without `wakeup`